test/logic/stake_logic_test.rb

```diff
@@ -57,6 +59,22 @@ def setup
     end
   end
 
+  test "get_stake_accounts \
+        when getting no response \
+        should raise error" do
+    authority = 'H2qwtMNNFh6euD3ym4HLgpkbNY6vMdf5aX5bazkU4y8b'
+    create(:stake_pool, authority: authority, network: 'testnet')
+
+    SolanaCliService.stub(:request, [], ['stakes', @testnet_url]) do
+      p = Pipeline.new(200, @initial_payload)
+                  .then(&get_last_batch)
+                  .then(&get_stake_accounts)
+
+      assert_nil p[:payload][:stake_accounts]
+      assert_equal 500, p.code
```

Comment:
This test does not reach the raise statement, it fails in cli_request. 
```
[1] pry(#<StakeLogicTest>)> p.errors
=> #<NoMethodError: undefined method `encode' for []:Array>
```

Add assertion for error message to be sure that it returns what you want.